docs/enhancements/cluster_pool_hot_spares.md

```diff
@@ -0,0 +1,111 @@
+# Maintaining Active Clusters in a ClusterPool
+
+[HIVE-1576](https://issues.redhat.com/browse/HIVE-1576)
+
+- [Maintaining Active Clusters in a ClusterPool](#maintaining-active-clusters-in-a-clusterpool)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [User Story](#user-story)
+  - [Proposals](#proposals)
+    - [ClusterClaim Prioritizes Active Clusters](#clusterclaim-prioritizes-active-clusters)
+    - [ClusterClaim Must Account For `hibernateAfter`](#clusterclaim-must-account-for-hibernateafter)
+    - [Don't Hibernate New ClusterDeployments](#dont-hibernate-new-clusterdeployments)
+    - [`activeCount`/`activePercent`](#activecountactivepercent)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Test Plan](#test-plan)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+Support configuring ClusterPools such that some number of clusters (0 <= N <= all) are Active for some amount of time (0 <= T <= eternity) before hibernating.
+
+## Motivation
+
+Today ClusterPools immediately hibernate new clusters.
+This means a ClusterClaim is guaranteed to suffer at least Resuming time before the cluster is available for use.
+While this is still faster than having to create a cluster from scratch, it is a potentially-unnecessary delay.
+As an example, CI runs could be reduced by ~5m if a ClusterClaim were able to grab an Active cluster rather than having to Resume a Hibernating one.
+
+### Goals
+
+Ability to tune a ClusterPool such that a ClusterClaim gets an Active cluster immediately in the common case.
+
+### Non-Goals
+
+Solving world peace.
+
+
+### User Story
+
+As a consumer of a (properly tuned) ClusterPool,
+
+I want ClusterClaim to give me an active cluster immediately in the common case,
+
+So that I can get my work done more quickly.
+
+(Ideally without sacrificing the cost savings of having some clusters in the pool hibernating.)
+
+## Proposals
+
+Several implementation alternatives have already come to mind.
+These are not necessarily mutually exclusive.
+
+### ClusterClaim Prioritizes Active Clusters
+
+For most/all of the options below, it makes sense to add logic to ClusterClaim to preferentially grab an Active cluster first.
```

Comment:
With sufficiently bad claim patterns this can lead to oldest hibernating clusters to not being claimed for prolonged time period. Not sure if that's a problem on its own, but we're seeing problems with clusters that spend too long in hibernation after installing (24h+), and claims not being FIFO could increase the odds of that.

Reply:
I'm curious what problems you've seen with old clusters getting stale.

There's a known bug (yet to be filed) where, if you change the ClusterPool's spec in ways that result in different ClusterDeployments, you're now lying about the CDs that were already in the pool. Is it that, or something else?

Reply:
No, it was the clusters massively restarting (thought to be cert rotation) after hibernated for 24h+: https://coreos.slack.com/archives/CE3ETN3J8/p1624379289071000?thread_ts=1624376710.056300&cid=CE3ETN3J8

Reply:
HIVE-1570

Reply:
Latest rev of the doc enforces strict FIFO by claiming in `creationTimestamp` order.